File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buyPrice=prices[0];
        int sellPrice=0;
        int profit=0;
        for(int i=1;i<prices.size();i++){
            sellPrice=prices[i];
            if(sellPrice<buyPrice){
                buyPrice=prices[i];
            }
            else{
                int diff=sellPrice-buyPrice;
                profit=max(profit,diff);
            }
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        buyTime=0
        maxprofit=0
        for i in range(1,len(prices)):
            if prices[i]<prices[buyTime]:
                buyTime=i
            else:
                maxprofit=max(maxprofit,prices[i]-prices[buyTime])
        return maxprofit